org.eclipse.scanning.command/src/org/eclipse/scanning/command/factory/ScanRequestExpresser.java

```diff
@@ -49,22 +50,41 @@ String pyExpress(ScanRequest<?> request, boolean verbose) throws Exception {
 			scanRequestPartiallyWritten |= true;
 		}
 
-		if (request.getMonitorNames() != null
-				&& request.getMonitorNames().size() > 0) {
+		if (request.getMonitorNamesPerPoint() != null
+				&& request.getMonitorNamesPerPoint().size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)